src/routes/conversation/[id]/+server.ts

```diff
@@ -14,12 +14,28 @@ import type { TextGenerationStreamOutput } from "@huggingface/inference";
 import { error } from "@sveltejs/kit";
 import { ObjectId } from "mongodb";
 import { z } from "zod";
+import { MESSAGES_RATE_LIMIT } from "$env/static/private";
+import { ERROR_MESSAGES } from "$lib/stores/errors.js";
 
 export async function POST({ request, fetch, locals, params }) {
 	const id = z.string().parse(params.id);
 	const convId = new ObjectId(id);
 	const date = new Date();
 
+	// rate limit users by messages per day
+	if (locals.user && MESSAGES_RATE_LIMIT) {
+		const messagesCount = await collections.messageEvents.countDocuments({
+			userId: locals.user._id,
+		});
+
+		if (messagesCount + 1 > parseInt(MESSAGES_RATE_LIMIT)) {
+			return new Response(JSON.stringify({ error: ERROR_MESSAGES.overRateLimit }), {
+				status: 429,
+				headers: { "Content-Type": "application/json" },
```

Comment:
@coyotte508 FYI I had to do this instead of throwing an error because of the way errors are handle in hf.js [there](https://github.com/huggingface/huggingface.js/blob/main/packages/inference/src/tasks/custom/streamingRequest.ts#L33). Do you think hf.js should forward the error instead of changing it?